addons/mass_mailing/models/mailing.py

```diff
@@ -640,6 +640,8 @@ def action_test(self):
         }
 
     def action_launch(self):
+        if self.mailing_on_mailing_list and not self.contact_list_ids:
+            return self._get_missing_mailing_list_warning()
```

Comment:
We could add a required in view but it is not the purpose of main api methods to return warnings ... it can be called in other situations, and is meant to be working on a batch recordset

Reply:
Yes I see, I tried to make sure different flows were covered but it may not be enough / appropriate.

However, the idea of this commit was precisely to remove the required attribute in order to be able to save the form (and all the work done on html edition) even if no list is set. So just to be sure, your suggestion is to keep the view as is? Thx for clarification :+1: 

Would there be another solution to implement in the template (not touching actions) ?

Reply:
Ah indeed, looked at the whole diff and not commit by commit :) . Indeed I prefer globally to keep the required on view. Contact list is a required field and it makes sense to prevent from saving an invalid mailing. I don't think it is a good idea to add business (py) code just because of some view related issues.

We can also remove the required on the field completely and that's it. If you send with a void contact_list_ids, no mails are generated and that's ok. If you send a mailing with a domain leading to 0 contacted records that's ok, you have 0 traces and the mailing is done. We could say it is the case here: if you send without lists, you will have 0 traces and the mailing is done. That is what you wanted :) .